lib/Dialect/TTNN/Workarounds/Broadcast.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "Broadcast.h"
```

Comment:
Couple of thoughts on this topic:
- I think we need to move larger transforms into separate subfolders inside `Transforms` folder, and to separate different passes and patterns into files. So for example we could have a `Transforms\Optimizer` folder where `Optimizer.cpp` could be split into multiple files if the logic there grows too large and it makes sense to split. We should then have a `Transforms\Workarounds` folder where we can move the `TTNNWorkarounds.cpp` and create separate files/subfolders for different workaround passes and patterns. So for example we could have separate subfolders for layout and for decomposition passes/patterns, so `Broadcast.cpp` would be under `Transforms\Workarounds\Decomposition` and other patterns that are currently implemented in `TTNNWorkarounds.cpp` should be extracted there or into `Transforms\Workarounds\Layout`. With this, I don't think we need to move the `Workarounds` outside of `Transforms` folder, they will be encapsulated in their own subfolder and they technically are transforms.
- Headers should follow the same folder structure inside `tt-mlir/include` folder, and `Broadcast.h` should be moved there.
- File names should be more telling, so instead of `Broadcast.h/cpp` it should be `BroadcastRewritePattern.h/cpp` or at least `BroadcastPattern.h/cpp`.

Reply:
I have another thought here that is a bit subjective and less relevant to this PR, but it is related. I know that llvm mlir keeps transforms inside each dialect folder, but to me that is very unintuitive, especially because they also have one `Transforms` folder in the root where they keep only common transforms. I think that inside `Dialect` folder we should keep only stuff regarding dialects modelling (their definitions, ops, attributes, interfaces, verifiers, etc). I think we should extract `Transforms` folder to be the same level as the `Dialect` folder, like [stablehlo](https://github.com/openxla/stablehlo/tree/main/stablehlo) does, and then inside `Transforms` folder we can have common transforms in the root, and subfolders per each dialect with their transforms. This is just my opinion, based on experience where I couldn't find the optimizer logic because it was buried in `Dialect` folder :)